File: LeetcodeV2/V2_python/59.py
class Solution:
    def generateMatrix(self, n: int) -> list[list[int]]:
        matrix = [[0] * n for _ in range(n)]
        
        left, right = 0, n - 1
        top, bottom = 0, n - 1
        
        cur = 1
        target = n * n
        
        while cur <= target:
            for i in range(left, right + 1):
                matrix[top][i] = cur
                cur += 1
            top += 1 
            
            for i in range(top, bottom + 1):
                matrix[i][right] = cur
                cur += 1
            right -= 1 

            for i in range(right, left - 1, -1):
                matrix[bottom][i] = cur
                cur += 1
            bottom -= 1 

            for i in range(bottom, top - 1, -1):
                matrix[i][left] = cur
                cur += 1
            left += 1 
            
        return matrix


if __name__ == "__main__":
    sol = Solution()
    
    print("n = 3:")
    res3 = sol.generateMatrix(3)
    for row in res3: print(row)
    
    print("\nn = 1:")
    res1 = sol.generateMatrix(1)
    for row in res1: print(row)

File: LeetcodeV2/V2_Java/59.java
import java.util.Arrays;
class Solution59 {
    public int[][] generateMatrix(int n) {
        int [][] res=new int[n][n];

        int cur=1;

        int top=0;
        int bottom=n-1;
        int left=0;
        int right=n-1;

        while(cur<= n*n){
            for(int i =left; i<=right;i++){
                res[top][i]=cur++;
                
            }
            top++;
            for(int i=top;i<=bottom;i++){
                res[i][right]=cur++;
                
            }
            right--;
            for(int i=right;i>=left;i--){
                res[bottom][i]=cur++;
                
            }
            bottom--;
            for(int i=bottom;i>=top;i--){
                res[i][left] = cur++;
            }
            left++;
        }

    return res;
    }

        public static void main(String[] args) {
        Solution59 sol = new Solution59();


        System.out.println("n = 3 的结果：");
        printMatrix(sol.generateMatrix(3));


        System.out.println("\nn = 1 的结果：");
        printMatrix(sol.generateMatrix(1));
    }


    private static void printMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            System.out.println(Arrays.toString(row));
        }
    }
}
